src/services/chain/apis/extrinsic.ts

```diff
@@ -95,46 +110,65 @@ export const createAccountViaService = async (
     data
   );
 
-  // nonce: -1 is needed in latest versions - see
-  // https://substrate.stackexchange.com/questions/1776/how-to-use-polkadot-api-to-send-multiple-transactions-simultaneously
+// https://substrate.stackexchange.com/questions/1776/how-to-use-polkadot-api-to-send-multiple-transactions-simultaneously
   extrinsic
     ?.signAndSend(serviceKey, { nonce: -1 }, ({ status, events }) => {
-      if (status.isInBlock) {
-        console.log(`Completed at block hash #${status.asInBlock.toString()}`);
-      } else {
-        console.log(`Current status: ${status.type}`);
-      }
-      callback(status, events);
+      const extrinsicFailedData = checkForFailedEvent(events);
+      extrinsicFailedData === undefined ? callback(status, events) : errorCallback(Error(extrinsicFailedData));
     })
     .catch((error: any) => {
       errorCallback(error);
     });
 };
 
-export const createMsaForProvider = async (
+export const createProviderMsa = async (
   callback: DsnpCallback,
   errorCallback: DsnpErrorCallback
 ) => {
   const api = requireGetProviderApi();
   const serviceKeys: KeyringPair = requireGetServiceKeys();
+
   // instantiate the extrinsic object
-  const extrinsic = api.tx.msa.create();
-  await extrinsic
+  const createMsaExtrinsic = api.tx.msa.create();
+  await createMsaExtrinsic
     ?.signAndSend(serviceKeys, { nonce: -1 }, ({ status, events }) => {
-      callback(status, events);
+      const extrinsicFailedData = checkForFailedEvent(events);
+      extrinsicFailedData === undefined ? callback(status, events) : errorCallback(Error(extrinsicFailedData));
     })
     .catch((error: any) => {
       errorCallback(error);
     });
+
 };
 
+export const registerProvider = async (
+    callback: DsnpCallback,
+    errorCallback: DsnpErrorCallback
+) => {
+  const api = requireGetProviderApi();
+  const serviceKeys = requireGetServiceKeys();
+
+  const createProviderExtrinsic = api.tx.msa.createProvider("ExClPr");
+
+  await createProviderExtrinsic
+      ?.signAndSend(serviceKeys, { nonce: -1 }, ({ status, events }) => {
+        const extrinsicFailedData = checkForFailedEvent(events);
+        console.log({extrinsicFailedData})
+        extrinsicFailedData === undefined ? callback(status, events) : errorCallback(Error(extrinsicFailedData));
+      })
+      .catch((error: any) => {
+        errorCallback(error);
+      });
+}
+
 export const fetchAllSchemas = async (): Promise<Array<SchemaDetails>> => {
   const api = requireGetProviderApi();
 
-  const schema_id = await (await api.query.schemas.schemaCount()).toNumber();
+  const schemasMaxHex = (await (await api.query.schemas.currentSchemaIdentifierMaximum())).toHex();
```

Comment:
`toNumber` was failing to build saying there is no such method. I just chose something easy

Reply:
Doesn't it come back as a bigint?

Reply:
apparently I hadn't checked that you could call "toBigInt" but it's fixed now.